### PR TITLE
HW-55799 throw error on transfer type not supported

### PR DESCRIPTION
**Pull request by peter-joseph**

### CODE REVIEW

File: transfermethodrepository/src/main/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodRepositoryImpl.java [hunk 1]
```diff
@@ -288,4 +295,12 @@ public Handler getHandler() {
             }
         });
     }
+
+    // Note: This way of surfacing error is not ideal but rather a workaround please have a look on other options,
+    // before resulting into this pattern
+    private HyperwalletErrors getErrorsOnUnsupportedTransferType() {
+        HyperwalletError error = new HyperwalletError(R.string.error_unsupported_transfer_type,
+                EC_UNEXPECTED_EXCEPTION);
+        return new HyperwalletErrors(Arrays.asList(error));
```
Comment (manikandan-altmtrk) [inline transfermethodrepository/src/main/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodRepositoryImpl.java]: Suggestion: we could simplify more.

Comment (peter-joseph) [reply transfermethodrepository/src/main/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodRepositoryImpl.java]:
```
return new HyperwalletErrors(Arrays.asList(
        new HyperwalletError(R.string.error_unsupported_transfer_type, EC_UNEXPECTED_EXCEPTION)));
```
you mean that one? ☝️, looks unreadable to me, if we are going that path we can also do the following:
```
default: // error on unknown transfer type
      callback.onError(new HyperwalletErrors(Arrays.asList(
        new HyperwalletError(R.string.error_unsupported_transfer_type, EC_UNEXPECTED_EXCEPTION))));
```
but I rather make it cleaner and readable, easy to understand what's going on too, what do you think?

File: transfermethodrepository/src/test/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodRepositoryImplTest.java [hunk 2]
```diff
@@ -157,6 +159,23 @@ public Object answer(InvocationOnMock invocation) {
         assertThat(mErrorsArgumentCaptor.getValue().getErrors(), hasItem(error));
     }
 
+    @Test
+    public void testCreateTransferMethod_createTransferWithUnsupportedTransferType() {
```
Comment (fmattos-hw) [inline transfermethodrepository/src/test/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodRepositoryImplTest.java]: I think testCreateTransferMethod_withUnsupportedTransferMethodType would be better

File: transfermethodrepository/src/test/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodRepositoryImplTest.java [hunk 3]
```diff
@@ -222,6 +241,26 @@ public Object answer(InvocationOnMock invocation) {
         assertThat(mErrorsArgumentCaptor.getValue().getErrors(), hasItem(error));
     }
 
+    @Test
+    public void testDeactivateTransferMethod_createTransferWithUnsupportedTransferType() {
```
Comment (fmattos-hw) [inline transfermethodrepository/src/test/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodRepositoryImplTest.java]: maybe testDeactivateTransferMethod_withUnsupportedTransferMethodType

File: commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java [hunk 4]
```diff
@@ -86,7 +86,7 @@ public static void setInstance(HyperwalletInsight hyperwalletInsight) {
      */
     public void initialize(@NonNull final Context context, @NonNull final Configuration configuration) {
         Insight.initialize(context, configuration.getEnvironment(), configuration.getProgramToken(), SDK_VERSION,
-                configuration.getInsightApiUrl(), configuration.getUserToken());
+                configuration.getInsightApiUri(), configuration.getUserToken());
```
[the remark this answers is not in the document]
Comment (peter-joseph) [reply commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java]: yep it's already in development branch that's why this PR is failing

File: transfermethodrepository/src/main/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodRepositoryImpl.java (hunk 1 above)
Comment (manikandan-altmtrk) [reply transfermethodrepository/src/main/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodRepositoryImpl.java]: okay @peter-joseph

**Pull request merged**